src/Single.php

```diff
@@ -171,4 +229,32 @@ public static function takeWhile(iterable $data, callable $predicate): \Generato
             }
         }
     }
+
+    /**
+     * Return pairs of elements from given collection.
+     *
+     * Returns empty generator if given collection contains less than 2 elements.
+     *
+     * @template T
+     * @param iterable<T> $data
+     *
+     * @return \Generator<array{T, T}>
+     */
+    public static function eachPair(iterable $data): \Generator
```

Comment:
I think the more-commonly recognized name for this functionality is [pairwise](https://en.wikipedia.org/wiki/Pairwise). How about renaming to that?

Reply:
Done.